api/debug/debug.go

```diff
@@ -219,18 +225,23 @@ func (d *Debug) handleTraceCall(w http.ResponseWriter, req *http.Request) error
 }
 
 func (d *Debug) createTracer(name string, config json.RawMessage) (tracers.Tracer, error) {
-	if strings.TrimSpace(name) == "" {
-		return nil, fmt.Errorf("tracer name must be defined")
-	}
-	_, noTracers := d.allowedTracers["none"]
-	_, allTracers := d.allowedTracers["all"]
-
-	// fail if the requested tracer is not allowed OR if the "all" tracers code isn't active
-	if _, ok := d.allowedTracers[name]; noTracers || (!ok && !allTracers) {
-		return nil, fmt.Errorf("creating tracer is not allowed: %s", name)
+	tracerName := strings.TrimSpace(name)
+	// compatible with old API specs
+	if tracerName == "" {
+		tracerName = "structLoggerTracer" // default to struct log tracer
+	}
+
+	// if it's builtin tracers
+	if tracers.DefaultDirectory.Lookup(tracerName) {
+		_, allowAll := d.allowedTracers["all"]
+		// fail if the requested tracer is not allowed OR "all" not set
+		if _, allowed := d.allowedTracers[tracerName]; !allowAll && !allowed {
+			return nil, fmt.Errorf("creating tracer is not allowed: %s", name)
+		}
+		return tracers.DefaultDirectory.New(tracerName, config, false)
 	}
 
-	return tracers.DefaultDirectory.New(name, config, d.allowCustomTracer)
+	return tracers.DefaultDirectory.New(tracerName, config, d.allowCustomTracer)
```

Comment:
Line 235 indicates it only handles built-in tracers and line 244 is for custom tracer enabled scenario.

Reply:
>What happens if a node starts with --api-allowed-tracers="none" and there is a request for the call tracer?

Line 239

Reply:
Apologies I was thinking `--api-allowed-tracers="none" ` and request with `non-exisiting tracer`.
- `--api-allowed-tracers="none"`, request: `1234` 
  - Before : `"creating tracer is not allowed: %s"`
  - Now: `js-debugger: unable to create custom tracer`
  
But after some digging I see why we're making the distinction between inbuilt and custom tracers! This makes sense.
I wish that the distinction was more visible (because the `tracers.DefaultDirectory.New` is pretty loaded).

It might make sense perhaps something in the lines of :
```
if d.allowCustomTracer {
  tracers.DefaultDirectory.New(tracerName, config, d.allowCustomTracer)
  }
  ```
  
But I understand that the `DefaultDirectory.New` already covers a big part of this. Perhaps it could do with a refactor too ? I'd leave that to your consideration.

Reply:
It will falls to
``` golang
if d.allowCustomTracer {
  tracers.DefaultDirectory.New(tracerName, config, true)

return nil, errors.New('unspported tracer')
```

It's the behavior inside `DefaultDirectory.New`. So I would prefer leave it there.

In short
+ With custom tracer enabled, request tracer `1234` will return `js-debugger: unable to create custom tracer`.
+ With custom tracer disabled,  request tracer `1234` will return `unspported tracer`.

Reply:
I think it's unexpected that some part of the flag control happens in the API ( for built-in and `--api-allowed-tracers`), but the other part happens inside the Directory.New ( for custom tracers and `--api-allow-custom-tracer` ). 

But I feel like I'm nitpicking at this stage, even refactoring the Lookup function to avoid duplication of code won't really add that much value imo. lgtm :)